tools/pythonpkg/tests/fast/test_all_types.py

```diff
@@ -84,6 +86,253 @@ def test_fetchall(self, duckdb_cursor):
+                ],
+                mask=[0, 0, 1],
+                dtype=object,
+            ),
+            'double_array': np.ma.array(
+                [
+                    [],
+                    [42.0, float('nan'), float('inf'), float('-inf'), None, -42.0],
+                    None,
+                ],
+                mask=[0, 0, 1],
+                dtype=object,
+            ),
+            'nested_int_array': np.ma.array(
+                [
+                    [],
+                    [[], [42, 999, None, None, -42], None, [], [42, 999, None, None, -42]],
+                    None,
+                ],
+                mask=[0, 0, 1],
+                dtype=object,
+            ),
+            'struct': np.ma.array(
+                [
+                    {'a': None, 'b': None},
+                    {'a': 42, 'b': '🦆🦆🦆🦆🦆🦆'},
+                    None,
+                ],
+                mask=[0, 0, 1],
+                dtype=object,
+            ),
+            'struct_of_arrays': np.ma.array(
+                [
+                    {'a': None, 'b': None},
+                    {'a': [42, 999, None, None, -42], 'b': ['🦆🦆🦆🦆🦆🦆', 'goose', None, '']},
+                    None,
+                ],
+                mask=[0, 0, 1],
+                dtype=object,
+            ),
+            'array_of_structs': np.ma.array(
+                [
+                    [],
+                    [{'a': None, 'b': None}, {'a': 42, 'b': '🦆🦆🦆🦆🦆🦆'}, None],
+                    None,
+                ],
+                mask=[0, 0, 1],
+                dtype=object,
+            ),
+            'map': np.ma.array(
+                [
+                    {'key': [], 'value': []},
+                    {'key': ['key1', 'key2'], 'value': ['🦆🦆🦆🦆🦆🦆', 'goose']},
+                    None,
+                ],
+                mask=[0, 0, 1],
+                dtype=object,
+            ),
+            'time': np.ma.array(
+                ['00:00:00', '23:59:59.999999', None],
+                mask=[0, 0, 1],
+                dtype=object,
+            ),
+            'time_tz': np.ma.array(
+                ['00:00:00', '23:59:59.999999', None],
+                mask=[0, 0, 1],
+                dtype=object,
+            ),
+        }
+
+        # The following types don't have a numpy equivalent, and are coerced to
+        # floating point types by fetchnumpy():
+        # - 'hugeint'
+        # - 'dec_4_1'
+        # - 'dec_9_4'
+        # - 'dec_18_6'
+        # - 'dec38_10'
+
+        # The following types lead to errors:
+        # Conversion Error: Date out of range in timestamp conversion
+        # - 'timestamp_array'
+        # - 'timestamptz_array'
+        # Conversion Error: Could not convert Timestamp(US) to Timestamp(NS)
+        # - 'timestamp'
+        # - 'timestamp_s'
+        # - 'timestamp_ms'
+        # - 'timestamp_tz'
+        # SystemError: <built-in function __import__> returned a result with an exception set
```

Comment:
This definitely sounds like something we should investigate further

Reply:
It's unrelated to the PR:

```
Python 3.9.13 | packaged by conda-forge | (main, May 27 2022, 16:56:21)
[GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import duckdb
>>> print(duckdb.__version__)
0.4.0
>>> con = duckdb.connect()
>>> t = con.table_function("test_all_types")
>>> t.project("date_array").execute().fetchnumpy()
SystemError: /home/conda/feedstock_root/build_artifacts/python-split_1653669925074/work/Objects/listobject.c:318: bad argument to internal function

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
SystemError: <built-in function __import__> returned a result with an error set
>>>
```

Might be appending a `NULL` to a Python list? Cf. https://github.com/python/cpython/blob/d348afa15d5a997e7a8e51c0f789f41cb15cc651/Objects/listobject.c#L318

Reply:
Yeah I didn't think it was related to this pr, just that we should investigate